buildchain/buildchain/salt_tree.py

```diff
@@ -81,158 +81,147 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
-    Path('salt/metalk8s/bootstrap/kubernetes_client.sls'),
-    Path('salt/metalk8s/bootstrap/mark_control_plane.sls'),
-    Path('salt/metalk8s/bootstrap/precheck.sls'),
-    Path('salt/metalk8s/bootstrap/preflight.sls'),
-    Path('salt/metalk8s/bootstrap/registry.sls'),
-    Path('salt/metalk8s/bootstrap/salt_master.sls'),
-
-    Path('salt/metalk8s/calico/configured.sls'),
-    Path('salt/metalk8s/calico/deployed.sls'),
-    Path('salt/metalk8s/calico/init.sls'),
-    Path('salt/metalk8s/calico/installed.sls'),
-
-    Path('salt/metalk8s/containerd/configured.sls'),
-    Path('salt/metalk8s/containerd/init.sls'),
-    Path('salt/metalk8s/containerd/installed.sls'),
+    Path('salt/metalk8s/container-engine/containerd/configured.sls'),
+    Path('salt/metalk8s/container-engine/containerd/init.sls'),
+    Path('salt/metalk8s/container-engine/containerd/installed.sls'),
 
     Path('salt/metalk8s/defaults.yaml'),
 
-    Path('salt/metalk8s/kubeadm/init/certs/apiserver.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/apiserver-etcd-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/apiserver-kubelet-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/ca.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-ca.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-deploy-ca-cert.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-healthcheck-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-peer.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/etcd-server.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/front-proxy-ca.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/front-proxy-deploy-ca-cert.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/front-proxy-client.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/installed.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/sa.sls'),
-    Path('salt/metalk8s/kubeadm/init/certs/sa-deploy-pub-key.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/control-plane/apiserver.sls'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/controller-manager.sls'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/files/htpasswd'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/files/manifest.yaml'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/control-plane/scheduler.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/etcd/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/etcd/files/manifest.yaml'),
-    Path('salt/metalk8s/kubeadm/init/etcd/local.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/init.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/kubelet-start/configured.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubelet-start/files/kubeadm.env'),
+    Path('salt/metalk8s/_internal/certs/init.sls'),
+    Path('salt/metalk8s/_internal/certs/installed.sls'),
+    Path('salt/metalk8s/_internal/certs/removed.sls'),
```

Comment:
I actually prefer `absent`. The *idempotent state* is "It doesn't exist" (i.e. *absent*), it is not (necessarily) *removed*. Also applies to other occurrences of `removed`.